src/jit/emitxarch.cpp

```diff
@@ -496,6 +496,95 @@ bool emitter::TakesVexPrefix(instruction ins)
     return IsAVXInstruction(ins);
 }
 
+
+//------------------------------------------------------------------------------------------
+// HasReservedVvvvBits: Checks if VEX or EVEX encoding of instruction has reserved VEX.vvvv
+//                      bits which must be set to 1111 otherwise instruction will #UD.
+//
+// Arguments:
+//    instruction -- processor instruction to check
+//
+// Return Value:
+//    true if instruction has a (E)VEX encoding with (E)VEX.vvvv bits reserved
+//
+bool emitter::HasReservedVvvvBits(instruction ins, code_t code)
```

Comment:
I still don't think this is the right fix. This only came up for `vpermpd` and `vpermq`, and was not a problem for the other ~60 instructions.

I think @fiigii's analysis that it was caused by adding `vpermpd` and `vpermq` to the `IsDstDstSrc` method is correct.

Reply:
I agree that fix for `vpermpd` and `vpermq` is to remove it from `IsDstDstSrc..` check, however, due to the fact that improperly set reserved `VEX.vvvv` bits cause #UD exception we should have such check at the place where those bits could be set.

Reply:
I don't think that adding another method to check that is the correct way to do it. These methods (like `IsDstDstSrc`) are already fairly expensive (I saw them taking up to 10% of the emit time, which was on par with `emitInsBinary`).

If an assert is desirable, it should probably be an assert, and should be checked via a flag (which will be possible after I finish the work on https://github.com/dotnet/coreclr/issues/19498).

Reply:
Agree with the arguments and that assert is enough to get it double checked.